### DISCUSSION TITLE
Image Quality

### DISCUSSION THREAD

**Discussion by issck** [stephanrauh/ngx-extended-pdf-viewer#2501] [Q&A]
Hi,

I'm trying the library to display PDFs as Blob in a modal. The issue is that we noticed the image/chart shown in the viewer is not as good as in the original PDF.
In the Actual Size zoom, the text is almost not recognizable. (if we download the pdf thru the viewer, and opens it by Adobe or browser, it shows fine)
![image](https://github.com/user-attachments/assets/da4d0fc6-4024-4efc-b0a7-826745505736)

And the image in Firefox looks better than in Chrome.

The image itself is not that big, I see in the document it mentions
"ngx-extended-pdf-viewer automatically reduces the resolution if necessary. You can also influence the resolution yourself. In the browser, it's determined by the attribute [(zoom)] and in print it's determined by `[printResolution]`."

Can you elaborate how to use zoom to enhance the resolution, or provide any suggestions. We set zoom="auto" in the app.

Thanks in advance,
Colin

**Comment by stephanrauh** [comment]
To be honest, I'm not sure I understand your question. Let's try to analyze it piece by piece.

I understand you've got a problem with rendering quality. You've also added a tiny screenshot that's indeed hardly legible.

What I do not understand is what you're referring to when you say "image". Is it an image embedded in the PDF file? Or are you talking about the PDF view itself?

Technically speaking, the PDF you're seeing in the viewer is an image drawn on a `<canvas>`. Basically, the resolution of the canvas matches the natural resolution of your screen. One pixel of the canvas is displayed as exactly one pixel on your screen (unless your browser zoom and/or your OS zoom isn't 100%). There's no option to change that, because it seems to be the natural choice.

The quote you've found refers to pathological corner cases. Some PDF files have an incredible resolution. If the resolution of the PDF file exceeds the capabilities of your browser, the viewer trades quality for reliability. But I'm pretty sure that's not your problem. Even the worst devices offer a 4K quality (see https://jhildenbiddle.github.io/canvas-size/#/?id=desktop).

My best guess is you've hit a bad combination of the resolution of an embedded image and the browser's resolution. This can result in mediocre rendering as your screenshot indicates. But that's only a wide shot.

BTW, you should be able to reproduce the issue in the native Firefox PDF viewer. I've modified pdf.js a lot, but I didn't touch the rendering engine, so both viewers should render identical results. Can you reproduce your issue in native Firefox or in the official pdf.js showcase (https://mozilla.github.io/pdf.js/web/viewer.html)?

**Comment by issck** [comment]
Thank you for your response.

The issue seems get resolved after I upgraded to "ngx-extended-pdf-viewer": "^21.1.2", was using 20.2.0.

The issue I had, was about an embedded image in the PDF file, which is a chart. The text in the chart are barely legible in the actual size or auto zoom. But somehow they are much clearer in Firefox.

Thanks again.
